x-pack/libbeat/autodiscover/providers/nomad/nomad.go

```diff
@@ -243,19 +240,29 @@ func (p *Provider) generateHints(event bus.Event) bus.Event {
 
 	rawMeta, ok := event["meta"]
 	if ok {
-		meta = rawMeta.(mapstr.M)
+		meta, ok = rawMeta.(mapstr.M)
+		if !ok {
+			goto postMeta
+		}
 		if nomadMeta, ok := meta["nomad"]; ok {
-			meta = nomadMeta.(mapstr.M)
+			meta, ok = nomadMeta.(mapstr.M)
+			if !ok {
+				goto postMeta
+			}
 		}
 
 		// The builder base config can configure any of the field values of nomad if need be.
 		e["nomad"] = meta
 		if rawAnn, ok := meta["tags"]; ok {
-			tags = rawAnn.(mapstr.M)
+			tags, ok = rawAnn.(mapstr.M)
+			if !ok {
+				goto postMeta
```

Comment:
nit: Is this part of this PR? Why decided to use goto?
Maybe to use an if/else implementation?